orm/main.go

```diff
@@ -63,6 +63,40 @@ func createOrmServer() *martini.ClassicMartini {
 		writeChallenge(rs, challenge)
 	})
 
+	m.Put("/challenge/:id", binding.MultipartForm(ChallengeForm{}), func(rs http.ResponseWriter, rq *http.Request, f ChallengeForm, params martini.Params) {
+		id := params["id"]
+		challenge, err := orm.FindChallenge(id)
+
+		challenge.Title = f.Title
+		challenge.TimeLimit = f.TimeLimit
+		challenge.MemoryLimit = f.MemoryLimit
```

Comment:
You should only update fields that were not empty from the put request.

Reply:
Just added a PATCH route that has this behavior, updates only the non-empty parameters.
Left this PUT route as it is, assuming everything shall be updated. 
How does that sound?